### PR TITLE
Tests covering the All reviews page components

### PR DESCRIPTION
**Pull request by AlexandraMoga**
A set of tests that are checking the "All reviews" page functionality - like filtering by score, flagging reviews, the list of user reviews.

### CODE REVIEW

File: tests/test_ratings.py [hunk 1]
```diff
@@ -164,3 +166,112 @@ def test_delete_rating(selenium, base_url, variables):
...
+    extension = variables['all_scores_addon']
+    selenium.get(f'{base_url}/addon/{extension}')
+    addon = Detail(selenium, base_url).wait_for_page_to_load()
+    addon.login('regular_user')
+    reviews = addon.ratings.click_all_reviews_link()
+    flag = reviews.review_items
+    # the 'Flag' menu is displayed only for reviews with text
+    # iterating through the list of reviews until a review with text is found
+    count = 0
+    while True:
```
Comment (jrbenny35) [inline tests/test_ratings.py:236]: Is this supposed to be ```while count``` ?

File: tests/test_ratings.py [hunk 2]
```diff
@@ -164,3 +166,112 @@ def test_delete_rating(selenium, base_url, variables):
...
+def test_flag_review_requires_login(selenium, base_url, variables):
+    extension = variables['detail_extension_slug']
+    selenium.get(f'{base_url}/addon/{extension}')
+    addon = Detail(selenium, base_url).wait_for_page_to_load()
+    reviews = addon.ratings.click_all_reviews_link()
+    review_item = reviews.review_items
+    # the 'Flag' menu is displayed only for reviews with text
+    # iterating through the list of reviews until a review with text is found
+    count = 0
+    while True:
```
Comment (jrbenny35) [inline tests/test_ratings.py:271]: Is this supposed to be ```while count``` ?

**Review by jrbenny35 — CHANGES_REQUESTED** [review]

File: tests/test_ratings.py (hunk 1 above)
Comment (AlexandraMoga) [reply tests/test_ratings.py:236]: @jrbenny35 If I'm using `while count` the code inside the while loop will not execute (it is not reached). With `while True` the code is doing what is supposed to, i.e. iterate through the list of reviews until the if condition is satisfied (i.e. a rating with text is found).

File: tests/test_ratings.py (hunk 2 above)
Comment (AlexandraMoga) [reply tests/test_ratings.py:271]: same as above

File: tests/test_ratings.py (hunk 1 above)
Comment (jrbenny35) [reply tests/test_ratings.py]: Any way we could use the amount of reviews in the review list as a limit?

Comment (AlexandraMoga) [reply tests/test_ratings.py]: @jrbenny35 yes, this is a better solution - https://github.com/mozilla/addons-release-tests/pull/260/commits/f45751c475198741191587677362521105341440

File: tests/test_ratings.py [hunk 3]
```diff
@@ -164,3 +166,112 @@ def test_delete_rating(selenium, base_url, variables):
...
+def test_flag_review_requires_login(selenium, base_url, variables):
+    extension = variables['all_scores_addon']
+    selenium.get(f'{base_url}/addon/{extension}')
+    addon = Detail(selenium, base_url).wait_for_page_to_load()
+    reviews = addon.ratings.click_all_reviews_link()
+    review_item = reviews.review_items
+    # the 'Flag' menu is displayed only for reviews with text
+    # iterating through the list of reviews until a review with text is found
+    count = 0
+    while count <= len(reviews.reviews_list):
```
Comment (jrbenny35) [inline tests/test_ratings.py:271]: YEESSSSSSSSSSS!!!!!

**Review by jrbenny35 — APPROVED** [review]
R+, GREAT work!

**Pull request merged**